src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1432,40 +1281,33 @@ public void SymmetricExceptWith(IEnumerable<T> other)
 
             if (asSorted != null && AreComparersEqual(this, asSorted))
             {
-                SymmetricExceptWithSameEC(asSorted);
+                SymmetricExceptWithSameComparer(asSorted);
             }
             else
             {
                 int length;
                 T[] elements = EnumerableHelpers.ToArray(other, out length);
                 Array.Sort(elements, 0, length, Comparer);
-                SymmetricExceptWithSameEC(elements, length);
+                SymmetricExceptWithSameComparer(elements, length);
             }
         }
 
-        private void SymmetricExceptWithSameEC(SortedSet<T> other)
+        private void SymmetricExceptWithSameComparer(SortedSet<T> other)
         {
             Debug.Assert(other != null);
             Debug.Assert(AreComparersEqual(this, other));
 
             foreach (T item in other)
             {
-                //yes, it is classier to say
-                //if (!this.Remove(item))this.Add(item);
-                //but this ends up saving on rotations                    
-                if (Contains(item))
-                {
-                    Remove(item);
-                }
-                else
-                {
-                    Add(item);
-                }
+                // Checking for Contains first saves on rotations compared to attempting to
+                // Remove first, then Add.
+                // TODO: Is that really true?
```

Comment:
Whoops, forgot to remove this TODO. This was more of a self-note than a question, but thanks for answering :)